### PR TITLE
[FIX] web_editor: fix commands in inline

### PR DESCRIPTION
**Pull request by sebgeelen**
Some commands from the powerBox command bar were not working properly in the e-shop product "terms and conditions" section.

This was due to the isolation of Odoo fields
inside the odoo editor as a all.
Those fields do not always have an editable block element to apply the command on.

We adapted some commands to be able to work in this context And we disable 
others commands that should not be apply in this context.

We also remove a redundant command (separator) in website pages.

task-2962067

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js [hunk 1]
```diff
@@ -208,8 +224,7 @@ export class Powerbox {
             }
             const term = this._lastText;
 
-            this._currentFilteredCommands =
-                term === '' ? this._currentOpenOptions.commands : await onValueChangeFunction(term);
+            this._currentFilteredCommands = await onValueChangeFunction(term);
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js:227]: Why the change ? Is it a bugfix ?

File: addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js [hunk 2]
```diff
@@ -320,14 +335,18 @@ export class Powerbox {
     // -------------------------------------------------------------------------
 
     _filter(term, commands) {
-        const initalCommands = commands;
+        const initalCommands = this.openedInInlineRoot ? commands.filter(c => c.allowInInlineRoot) : commands;
+        if (term === '') {
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js:339]: Looks like you moved the logic here ?

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js [hunk 3]
```diff
@@ -1930,6 +1930,7 @@ const Wysiwyg = Widget.extend({
                 title: _t('Link'),
                 description: _t('Add a link.'),
                 fontawesome: 'fa-link',
+                allowInInlineRoot: true,
```
Comment (dmo-odoo) [inline addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1933]: What I had in mind was more like `condition: <predicate>` and we would have a predicate util which would be is `isInBlock`, so that the commands definitions would read:
```js
condition: isInBlock,
```

File: addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js [hunk 4]
```diff
@@ -152,6 +157,17 @@ export class Powerbox {
                     !this._active &&
                     (!this.options.shouldActivate || this.options.shouldActivate())
                 ) {
+                    this.openedInInlineRoot = true;
```
[the remark this answers is not in the document]
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js:160]: sure

File: addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js (hunk 1 above)
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js:227]: we need to pass in the filter every time to ensure the commands who should not appear in inline context are filtered out.

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js (hunk 3 above)
Comment (sebgeelen) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1933]: I can try that indeed, it would be more extensible. 
I'm just not sure yet if this will be possible easely at command definition outside of the odooeditor scope.

will try

File: addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js (hunk 2 above)
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/powerbox/Powerbox.js:339]: see https://github.com/odoo/odoo/pull/106348#discussion_r1031262397

File: addons/web_editor/static/lib/odoo-editor/src/commands/commands.js [hunk 5]
```diff
@@ -659,7 +669,7 @@ export const editorCommands = {
             range.startContainer,
             'P, PRE, H1, H2, H3, H4, H5, H6, BLOCKQUOTE',
         );
-
+
```
[the remark this answers is not in the document]
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js:672]: oops

File: addons/website/static/src/js/menu/edit.js [hunk 6]
```diff
@@ -555,7 +555,7 @@ var EditPageMenu = websiteNavbarData.WebsiteNavbarActionWidget.extend({
             {
                 groupName: _t('Website'),
                 title: _t('Separator'),
-                description: _t('Insert an horizontal separator sippet.'),
+                description: _t('Insert an horizontal separator snippet.'),
```
Comment (rdeodoo) [inline addons/website/static/src/js/menu/edit.js:558]: Good catch :D
But pretty sure this one is a typo too
```suggestion
                description: _t('Insert a horizontal separator snippet.'),
```

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 7]
```diff
@@ -1923,11 +1915,24 @@ export class OdooEditor extends EventTarget {
                 title: this.options._t('Switch direction'),
                 description: this.options._t('Switch the text\'s direction.'),
                 fontawesome: 'fa-exchange',
+                allowInInlineRoot: true,
                 callback: () => {
                     this.execCommand('switchDirection');
                 },
             },
         ];
+        if(!this.options.commands.filter(c =>  c.title === this.options._t('Separator')).length) {
```
Comment (Zinston) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:1924]: I'm gonna be annoying and point out the missing space after `if` and the extra space before `c.title`. There.

File: addons/web_editor/static/src/js/wysiwyg/wysiwyg.js (hunk 3 above)
Comment (Zinston) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1933]: Are you sure this can't be done with the already existing [`isDisabled` property](https://www.odoo.com/documentation/16.0/developer/reference/frontend/odoo_editor.html#reference) ?
By the way if you change the API, please make sure to update the doc, it would be too bad for it to be outdated.

Comment (sebgeelen) [reply addons/web_editor/static/src/js/wysiwyg/wysiwyg.js:1933]: oh yes, didn't knew this was a thing. I'll use that.

File: addons/web_editor/static/lib/odoo-editor/src/commands/commands.js [hunk 8]
```diff
@@ -496,13 +497,22 @@ export const editorCommands = {
                 node.remove();
             } else {
                 const block = closestBlock(node);
-                if (!['OL', 'UL'].includes(block.tagName)) {
-                    const ublock = block.closest('ol, ul');
-                    ublock && getListMode(ublock) == mode ? li.add(block) : blocks.add(block);
+                if (block.classList.contains('o_editable') ||
+                    ['OL', 'UL'].includes(block.tagName)
+                ) {
+                    continue;
```
[the remark this answers is not in the document]
Comment (sebgeelen) [reply addons/web_editor/static/lib/odoo-editor/src/commands/commands.js]: I removed this change after we changed the approach. I don't thing it's necessary to keep this check.

File: addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js [hunk 9]
```diff
@@ -1843,6 +1843,7 @@ export class OdooEditor extends EventTarget {
                 title: this.options._t('Heading 1'),
                 description: this.options._t('Big section heading.'),
                 fontawesome: 'fa-header',
+                isDisabled: () => {return !this.isSelectionInBlockRoot()},
```
Comment (dmo-odoo) [inline addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js:1846]: I find it weird to have `() => {return ...}`. Either do:
```js
() => {
    return ...;
}
```
or, better in this case IMO:
```
() => ...,
```
There are obviously a lot of occurences of this in the diff.

**Comment by dmo-odoo** [comment]
@robodoo r+
